components_library/memory/abstract_memory_system.py

```diff
@@ -41,17 +41,33 @@ def __init__(self) -> None:
 
     @abstractmethod
     def incorporate_memory(self, board: AbstractBoard) -> None:
+        """This function completes all of the necessary steps to add this
+        memory system to the board."""
         raise NotImplementedError
 
     @abstractmethod
     def get_mem_ports(self) -> Sequence[Tuple[AddrRange, Port]]:
+        """Get the ports to connect this memory system to the cache"""
         raise NotImplementedError
 
     @abstractmethod
     def get_memory_controllers(self) -> List[MemCtrl]:
+        """Get all of the memory controllers in this memory system"""
         raise NotImplementedError
 
     @abstractmethod
-    def get_memory_ranges(self) -> List[AddrRange]:
```

Comment:
Surely we still want to be able to get the memory ranges?

Reply:
Why? I would argue that should be hidden since the board should be in charge of this.

Reply:
Fair point. I was battling over what should be in charge of memory ranges, but yes, the board is the place for it to be.